packages/http-client-java/emitter/src/emitter.ts

```diff
@@ -113,127 +113,80 @@ export const $lib = createTypeSpecLibrary({
 
 export async function $onEmit(context: EmitContext<EmitterOptions>) {
   const program = context.program;
-  const options = context.options;
-  if (!options["flavor"]) {
-    if (options["package-dir"]?.toLocaleLowerCase().startsWith("azure")) {
-      // Azure package
-      options["flavor"] = "azure";
+  await validateDependencies(program, true);
+
+  if (!program.hasError()) {
```

Comment:
Most change here is due to this added `!program.hasError()` check. If `validateDependencies` reports error, emitter will skip processing.